src/mixins/dropdown.js

```diff
@@ -379,7 +379,12 @@ export default {
     focusInHandler(evt) {
       const target = evt.target
       // If focus leaves dropdown, hide it
-      if (this.visible && !contains(this.$refs.menu, target) && !contains(this.toggler, target)) {
+      if (
+        this.visible &&
+        !contains(this.$refs.menu, target) &&
+        !contains(this.toggler, target) &&
+        !hasClass(target, 'dropdown-toggle')
```

Comment:
The only issue with this, is when keyboard-only users tab to a dropdown toggle button, hit eneter to "click" the menu open, then tab through the items in the menu until they get to the next tabable on the screen (which is another dropdown toggle).  The menu stays open and obscures the button (I've pushed the top dropdown toggle over a a bit to show the dropdown toggle below it that is obscured):

![image](https://user-images.githubusercontent.com/2781561/65375682-b43e3600-dc6e-11e9-9fbd-90afce57a2d0.png)


To works fine if users are clicking, but not if using tabs to navigate to the next dropdown.

We might need to add in additional checks, or maybe figure out why the trigger is not opening the next menu (in the original issue)

I

Reply:
Good catch

Reply:
Otherwise, it was working OK to fix the original issue.

Reply:
It might be a timing issue with the dropdowns and listening/emitting events (which cause the dropdown to close if another is open), or something similar.

Reply:
@tmorehouse Yeah, I think so too.

Reply:
Maybe we may need to add in a bunch of console log debug statements in each event handler of dropdown mixin, just to see what sequence of events is preventing the second dropdown in the vertical navbar from opneing when the first one is closed.

Or perhaps it is related to the `display` being flex-column in the original issue markup...

May need to nextTick delay a few things or use requestAF to delay a few things (to make sure the transition is completed)

Reply:
Hmmmm

Something interesting:

- Go to the fiddle: https://jsfiddle.net/28dr50bk/
- Click on "User" dropdown toggle to open it.
- Then click on "Lang" dropdown toggle to open it

Everything works as expected...

But click on "User" first, then "Lang" then the issue happens.

Reply:
What is one simply wraps the following line `this.visible = false` in a `$nextTick()`?

I think what is happening is that the toggle button is moving (because the top dropdown is collapsing), and is not really what was clicked to close the first dropdown?

Reply:
@tmorehouse Noticed that too. Also tried the `$nextTick()` but it didn't helped.

Reply:
Hmmmm... maybe a requestAF instead of nextTick (or a combination of nextTick + requestAF)?

Reply:
Something else interesting.... if using tab to navigate the navbar dropdowns, everything works as epected... it is only click on the bottom dropdown toggle (when first dropdown is open) that shows the issue.

Reply:
Just trying a delay on the visibility `show` handler to see if that makes a difference...